Find a functionally equivalent code:android.text.TextUtils.getLayoutDirectionFromLocale:COMMENT
Method Modifier: public      static      
Comment:/**
 * Return the layout direction for a given Locale
 *
 * @param locale the Locale for which we want the layout direction. Can be null.
 * @return the layout direction. This may be one of:
 * {@link android.view.View#LAYOUT_DIRECTION_LTR} or
 * {@link android.view.View#LAYOUT_DIRECTION_RTL}.
 *
 * Be careful: this code will need to be updated when vertical scripts will be supported
 */

@@ -1,6 +1,6 @@
 {
     if (locale != null && !locale.equals(Locale.ROOT)) {
-        final String scriptSubtag = ICU.getScript(ICU.addLikelySubtags(locale.toString()));
+        final String scriptSubtag = ICU.addLikelySubtags(locale).getScript();
         if (scriptSubtag == null)
             return getLayoutDirectionFromFirstChar(locale);
         if (scriptSubtag.equalsIgnoreCase(ARAB_SCRIPT_SUBTAG) || scriptSubtag.equalsIgnoreCase(HEBR_SCRIPT_SUBTAG)) {

